parser.y

```diff
@@ -4022,16 +4024,48 @@ CreateTableSelectOpt:
 	{
 		$$ = &ast.CreateTableStmt{}
 	}
-|	SetOprStmt1
+|	SetOprStmt
+	{
+		$$ = &ast.CreateTableStmt{Select: $1.(ast.ResultSetNode)}
+	}
+|	SelectStmt
+	{
+		$$ = &ast.CreateTableStmt{Select: $1.(ast.ResultSetNode)}
+	}
+|	SelectStmtWithClause
 	{
 		$$ = &ast.CreateTableStmt{Select: $1.(ast.ResultSetNode)}
 	}
+|	SubSelect
+	{
+		var sel ast.ResultSetNode
+		switch x := $1.(*ast.SubqueryExpr).Query.(type) {
+		case *ast.SelectStmt:
+			x.IsInBraces = true
+			sel = x
+		case *ast.SetOprStmt:
+			x.IsInBraces = true
+			sel = x
+		}
+		$$ = &ast.CreateTableStmt{Select: sel}
+	}
 
 CreateViewSelectOpt:
-	SetOprStmt1
-|	'(' SetOprStmt1 ')'
+	SetOprStmt
+|	SelectStmt
+|	SelectStmtWithClause
+|	SubSelect
 	{
-		$$ = $2
+		var sel ast.StmtNode
+		switch x := $1.(*ast.SubqueryExpr).Query.(type) {
+		case *ast.SelectStmt:
+			x.IsInBraces = true
+			sel = x
+		case *ast.SetOprStmt:
+			x.IsInBraces = true
+			sel = x
```

Comment:
```suggestion
		case *ast.SelectStmt, *ast.SetOprStmt:
			x.IsInBraces = true
			sel = x
```

Reply:
No, golang can't do this. In this statement, x will be regarded as `ResultNode`.

Reply:
No, golang can't do this. In this statement, x will be regarded as `ResultNode`.